src/plugins/intel_cpu/src/nodes/search_sorted.cpp

```diff
@@ -0,0 +1,114 @@
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void SearchSorted::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset15::SearchSorted.
+}
+
+void SearchSorted::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    ov::element::Type inputPrec = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type outputPrec = getOriginalOutputPrecisionAtPort(0);
+
+    addSupportedPrimDesc({{LayoutType::ncsp, inputPrec}, {LayoutType::ncsp, inputPrec}},
+                         {{LayoutType::ncsp, outputPrec}},
+                         impl_desc_type::ref);
+}
+
+bool SearchSorted::created() const {
+    return getType() == Type::SearchSorted;
+}
+
+bool SearchSorted::needPrepareParams() const {
+    return false;
+}
+
+void SearchSorted::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+template <typename INPUT_TYPE, typename OUTPUT_TYPE>
+void SearchSorted::executeImpl() {
+    ov::reference::search_sorted<INPUT_TYPE, OUTPUT_TYPE>(getSrcDataAtPortAs<const INPUT_TYPE>(0),
+                                                          getSrcDataAtPortAs<const INPUT_TYPE>(1),
+                                                          getDstDataAtPortAs<OUTPUT_TYPE>(0),
+                                                          ov::Shape{getSrcMemoryAtPort(0)->getStaticDims()},
+                                                          ov::Shape{getSrcMemoryAtPort(1)->getStaticDims()},
+                                                          right_mode);
+}
+
+namespace {
+struct SearchSortedContext {
+    SearchSorted& node;
+};
+}  // namespace
+
+template <typename T>
+struct SearchSorted::SearchSortedExecute {
+    using TInputType = typename std::tuple_element<0, T>::type;
+    using TOutputType = typename std::tuple_element<1, T>::type;
+
+    void operator()(SearchSortedContext& ctx) {
+        ctx.node.executeImpl<TInputType, TOutputType>();
+    }
+};
+void SearchSorted::execute(dnnl::stream strm) {
+    auto inputPrecision = getParentEdgeAt(0)->getMemory().getDesc().getPrecision();
+    auto outputPrecision = getChildEdgeAt(0)->getMemory().getDesc().getPrecision();
+
+    SearchSortedContext ctx = {*this};
+
+#define CASE(OV_TYPE)                                                                           \
+    OV_CASE2(OV_TYPE, ov::element::i64, ov::element_type_traits<OV_TYPE>::value_type, int64_t), \
+        OV_CASE2(OV_TYPE, ov::element::i32, ov::element_type_traits<OV_TYPE>::value_type, int32_t)
+
+    OV_SWITCH(intel_cpu,
+              SearchSortedExecute,
+              ctx,
+              std::tie(inputPrecision, outputPrecision),
+              CASE(ov::element::f32),
+              CASE(ov::element::f16),
+              CASE(ov::element::bf16),
+              CASE(ov::element::i32),
+              CASE(ov::element::u32),
```

Comment:
U32 is not needed since plugin converts it to i32

Reply:
Fixed.